bee-common/bee-packable/src/packable/vec_prefix.rs

```diff
@@ -8,117 +8,72 @@ use crate::{
     error::{PackError, PackPrefixError, UnpackError, UnpackPrefixError},
     packer::Packer,
     unpacker::Unpacker,
-    Packable,
+    Bounded, BoundedU16, BoundedU32, BoundedU64, BoundedU8, InvalidBoundedU16, InvalidBoundedU32, InvalidBoundedU64,
+    InvalidBoundedU8, Packable,
 };
 
 use alloc::vec::Vec;
 use core::{convert::TryFrom, marker::PhantomData};
 
-/// Error encountered when attempting to convert a [`Vec<T>`] into a [`VecPrefix`], where
-/// the length of the source vector exceeds the maximum length of the [`VecPrefix`].
-#[derive(Debug, PartialEq, Eq)]
-pub struct PrefixedFromVecError {
-    /// Maximum length of the [`VecPrefix`].
-    pub max_len: usize,
-    /// Actual length of the source vector.
-    pub actual_len: usize,
-}
-
-/// Wrapper type for [`Vec<T>`] where the length prefix is of type `P`.
-/// The [`Vec<T>`]'s maximum length is provided by `N`.
+/// Wrapper type for [`Vec<T>`] with a length prefix.
+/// The [`Vec<T>`]'s maximum length is provided by `B`, where `B` is a [`Bounded`] type.
```

Comment:
Maybe just say bounds instead of maximum ? It's weird to mention maximum but not minimum

Reply:
Yep, makes sense.